lua/barbar/events.lua

```diff
@@ -130,7 +130,7 @@ function events.enable()
       'WinEnter', 'WinLeave',
     },
     {
-      callback = function() render.update() end,
```

Comment:
Could you put `schedule_wrap` around this?  `render.update` takes arguments w/ defaults; iirc some will be passed from the autocmd as the event table which won't be in the form that the function expects

Reply:
If I remember correctly:

```lua
vim.schedule_wrap(foo) == function(...)
  vim.schedule(function() foo(...) end)
end
```

Since an `autocmd` takes a `callback` in the form `(event: table) => true | nil` (in TypeScript annotations), really what we get is this:

```lua
function(event)
  vim.schedule(function() render.update(event) end)
end
```

So the input that is getting passed through the `callback` to `update` (i.e. `event: table`) is not an expected data type for the argument in that position.

There are a couple of other places in the codebase where this pattern exists (e.g. [here](https://github.com/romgrk/barbar.nvim/blob/master/lua%2Fbarbar%2Fevents.lua#L238-L242)), and I've had to do it in [my config](https://gitlab.com/Iron_E/dotfiles/-/blob/4d6d154f7f9734bd02e7aa02e45df57f25f0fa06/.config/nvim/lua/init/config.lua#L123).

Reply:
Thanks for your feedback, I think I've made the requested change ;)